generated/apimanagement/microsoft.apimanagement/2017-03-01/types.md

```diff
@@ -674,12 +698,12 @@
 
 ## LoggerContractProperties
 ### Properties
-* **credentials**: [LoggerContractPropertiesCredentials](#loggercontractpropertiescredentials) (Required): The name and SendRule connection string of the event hub for azureEventHub logger.
+* **credentials**: [LoggerContractPropertiesCredentials](#loggercontractpropertiescredentials) (ReadOnly): The name and SendRule connection string of the event hub for azureEventHub logger.
 Instrumentation key for applicationInsights logger.
-* **description**: string: Logger description.
-* **isBuffered**: bool: Whether records are buffered in the logger before publishing. Default is assumed to be true.
-* **loggerType**: 'applicationInsights' | 'azureEventHub' | string (Required): Logger type.
-* **sampling**: [LoggerSamplingContract](#loggersamplingcontract): Sampling settings contract.
+* **description**: string (ReadOnly): Logger description.
+* **isBuffered**: bool (ReadOnly): Whether records are buffered in the logger before publishing. Default is assumed to be true.
+* **loggerType**: 'applicationInsights' | 'azureEventHub' | string (ReadOnly): Logger type.
+* **sampling**: [LoggerSamplingContract](#loggersamplingcontract) (ReadOnly): Sampling settings contract.
```

Comment:
The `Required` property flag is correct when a user creates a `Microsoft.ApiManagement/service/loggers@2017-03-01` resource (in which case this object is part of the PUT request), and the `ReadOnly` property flag is correct when a user creates a `Microsoft.ApiManagement/service/apis/diagnostics/loggers@2017-03-01` or `Microsoft.ApiManagement/service/diagnostics/loggers@2017-03-01` resources (in which cases the object is only part of the PUT response).

What's being generated here is not correct, but neither is what was generated last on the `main` branch. I think we need a better approach to shape reuse, potentially before adding types for readonly resources.